api/v1beta2/ibmpowervscluster_webhook.go

```diff
@@ -89,3 +96,32 @@ func (r *IBMPowerVSCluster) validateIBMPowerVSClusterNetwork() *field.Error {
 	}
 	return nil
 }
+
+func (r *IBMPowerVSCluster) validateIBMPowerVSClusterCreateInfraPrereq() *field.Error {
+	annotations := r.GetAnnotations()
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	value, found := annotations[CreateInfrastructureAnnotation]
+	if !found {
+		return nil
+	}
+
+	createInfra, err := strconv.ParseBool(value)
+	if err != nil {
+		return field.Invalid(field.NewPath("annotations"), r.Annotations, "value of powervs.cluster.x-k8s.io/create-infra should be boolean")
+	}
+
+	if createInfra {
```

Comment:
lets do early return
```suggestion
	if !createInfra {
	     return nil
	}
```

Reply:
Done